env0/resource_template.go

```diff
@@ -259,16 +259,19 @@ func resourceTemplateDelete(ctx context.Context, d *schema.ResourceData, meta in
 }
 
 func resourceTemplateImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
-	return nil, errors.New("Not implemented")
-	// apiClient := meta.(*client.ApiClient)
-
-	// id := d.Id()
-	// template, err := apiClient.Template(id)
-	// if err != nil {
-	// 	return nil, err
-	// }
-
-	// d.Set("name", template.Name)
-
-	// return []*schema.ResourceData{d}, nil
+	id := d.Id()
+	var getErr diag.Diagnostics
+	_, uuidErr := uuid.Parse(id)
+	if uuidErr == nil {
+		_, getErr = getTemplateById(id, meta)
+	} else {
+		var template client.Template
+		template, getErr = getTemplateByName(id, meta)
+		d.SetId(template.Id)
```

Comment:
We should log wether we query by name or id

Reply:
https://github.com/env0/terraform-provider-env0/pull/41/commits/3ad489ce9e5200d5392e614eb3535be90a676114